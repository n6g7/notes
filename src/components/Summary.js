import React, { PureComponent } from 'react'
import { Link } from 'react-router'

import notebooks from '../../notebooks'

class Summary extends PureComponent {
  render () {
    return <div>
      <h1>summary</h1>
      <ul>
        { Object.keys(notebooks).map(notebook =>
          <li key={`summary-${notebook}`}>
            <Link to={`/nb/${notebook}`}>{notebook}</Link>
          </li>
        ) }
      </ul>
    </div>
  }
}

export default Summary
