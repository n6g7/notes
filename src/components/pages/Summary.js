import React, { PureComponent } from "react"
import { Link } from "react-router-dom"

import notebooks from "@notebooks"
import { Page } from "@templates"

class Summary extends PureComponent {
  render() {
    return (
      <Page>
        <h1>summary</h1>
        <ul>
          {Object.keys(notebooks).map((notebook) => (
            <li key={`summary-${notebook}`}>
              <Link to={`/nb/${notebook}`}>{notebook}</Link>
            </li>
          ))}
        </ul>
      </Page>
    )
  }
}

export default Summary
