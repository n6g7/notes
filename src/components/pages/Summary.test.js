import React from 'react'
import { shallow } from 'enzyme'

import Summary from './Summary'

describe('<Summary />', () => {
  it('renders well', () => {
    const wrapper = shallow(<Summary />)
    expect(wrapper).toMatchSnapshot()
  })
})
