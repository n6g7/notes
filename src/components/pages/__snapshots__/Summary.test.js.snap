// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> renders well 1`] = `
<Page>
  <h1>
    summary
  </h1>
  <ul>
    <li
      key="summary-accelerating-returns"
    >
      <Link
        replace={false}
        to="/nb/accelerating-returns"
      >
        accelerating-returns
      </Link>
    </li>
    <li
      key="summary-bertrand"
    >
      <Link
        replace={false}
        to="/nb/bertrand"
      >
        bertrand
      </Link>
    </li>
    <li
      key="summary-secretary"
    >
      <Link
        replace={false}
        to="/nb/secretary"
      >
        secretary
      </Link>
    </li>
    <li
      key="summary-simulation"
    >
      <Link
        replace={false}
        to="/nb/simulation"
      >
        simulation
      </Link>
    </li>
    <li
      key="summary-star-death"
    >
      <Link
        replace={false}
        to="/nb/star-death"
      >
        star-death
      </Link>
    </li>
  </ul>
</Page>
`;
